### ISSUE TITLE
Agent spamming logs with stats error

### ISSUE THREAD

**Issue by jasonmoo** [aws/amazon-ecs-agent#530]
Experienced an event in our cluster where servers began running out of disk space.  Found several gigs of logs in `/var/logs/ecs/ecs-agent.log.YYYY-MM-DD-HH` comprising only a few hours of logs.  The `agent` appeared to be using 20-30% cpu in `top`. 

After clearing the logs and restarting the agent the server resumed normal function.  This happened across several servers in our cluster.  A possible external cause is exhausting server resources, however the cause within the agent is unclear.

Found several `closed pipe` errors like the one below preceeding the `too many open files` errors.  

Possible path to explore is: connection errors not resulting in connection close.

```
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: io: read/write on closed pipe
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
2016-09-16T09:31:42Z [WARN] Error retrieving stats for container 408359b3cc6dbe097edbcbbb68833b662da73b9990f52f1cac4ae447634f6d54: dial unix /var/run/docker.sock: socket: too many open files
...
```

**Comment by aaithal** [comment]
@jasonmoo Can you please let us know the ECS Agent version on the Instance where you're seeing this? This looks very similar to #515 and we have released a fix for it in the  [v1.12.2](https://github.com/aws/amazon-ecs-agent/releases/tag/v1.12.2) version of the ECS Agent.

> We have released the [v1.12.2](https://github.com/aws/amazon-ecs-agent/releases/tag/v1.12.2) version of the ECS Agent today, which should fix this issue. Please let us know if you continue running into this. Closing this issue for now. Feel free to reach out to us if you have any questions/comments/feedback.

The latest list of AMI IDs can be found [here](http://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-optimized_AMI.html)

Please let us know if upgrading to the new version solves the issue.
Thanks,
Anirudh

**Comment by jasonmoo** [comment]
Ah yes my mistake:  

```
ECS     Docker   AMI
1.11.1  1.11.2   amzn-ami-2016.03.f-amazon-ecs-optimized (ami-f3468e93)
```

Will upgrade and report back.
